src/file_watching.jl

```diff
@@ -1,56 +1,153 @@
+"A watched file; construct with its path as single argument"
+mutable struct WatchedFile
+    path::String
+    mtime::Float64
+    WatchedFile(path::String) = new(path, mtime(path))
+end
+
```

Comment:
You probably want `WatchedFile{T}  where T<:AbstractString` as  paths may be produced by a `replace` mechanism or whatever and then it's a `SubString` not a `String`.

Reply:
Overall, like this? Am not fully comfortable yet with parametric type stuff...
```julia
mutable struct WatchedFile{T<:AbstractString}
    path::T
    mtime::Float64
    WatchedFile(path) = new{typeof(path)}(path, mtime(path))
end
```

Reply:
The last one can just be `WatchedFile(path) = new(path, mtime(path))` it will pick up `T` on the fly.

Reply:
```julia
mutable struct WatchedFile{T<:AbstractString}
  path::T
  mtime::Float64
end
WatchedFile(path) = WatchedFile(path, mtime(path))
```

Reply:
Nope, will raise error "Too few type parameters specified in "new{...}". The internal constructor replaces the default constructor, which is intended here. Creating a `WatchedFile` with anything else than it's actual `mtime` does not make sense.

Reply:
Hmm that's old Julia behaviour, I tested before posting the code. What version are you running? Anyway doesn't really matter.

Reply:
I'm on 1.1.0... that changed since then?

Reply:
yeah not exactly sure, copy pasting in Julia 1.2:

```julia
julia> mutable struct WatchedFile{T<:AbstractString}
         path::T
         mtime::Float64
       end

julia> WatchedFile(path) = WatchedFile(path, mtime(path))
WatchedFile

julia> WatchedFile("/Users/tlienart/Desktop/testconv.jl")
WatchedFile{String}("/Users/tlienart/Desktop/testconv.jl", 1.548580231e9)
```